contracts/adapter/UniV3Adapter.sol

```diff
@@ -57,17 +57,26 @@ contract UniV3Adapter {
     }
 
     if (isETH(tokenOut)) {
-      weth.withdraw(address(this).balance);
+      uint wethBal = weth.balanceOf(address(this));
+      weth.withdraw(wethBal);
+      if (wethBal < tokenOutAmount) {
+        revert NotEnoughETH();
```

Comment:
Won't this revert at the limitSwapVerifier level? We could save some gas here by removing this and getting rid of the need for the wethBal variable

Reply:
it won't make it to the verifier, if this check isn't here it reverts on the next line with no revert message, since it's a failed ETH transfer. that's what was happening when we were getting revert without reason string. this gives us a chance of seeing a meaningful error message at least, and it's pretty minimal on gas